code/mapping/ext_modules/mapping_common/map.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from genpy.rostime import Time
+from std_msgs.msg import Header
+
+from mapping_common.entity import Entity
+
+from mapping import msg
+
+
+@dataclass
+class Map:
+    """2 dimensional map for the intermediate layer
+
+    General information:
+    - 2D top-down map. The height(z) dimension is mostly useless
+      for collision detection and path planning
+    - The map is based on the local car position and is only built using sensor data.
+      No global positioning via GPS or similar is used.
+    - The map (0/0) is the center of the hero car
+    - All transformations to entities are relative to
+      the hero car's coordinate system (position/heading)
+    - The map's x-axis is aligned with the heading of the hero car
+    - The map's y-axis points to the left of the hero car
+    - Coordinate system is a right-hand system like tf2 (can be visualized in RViz)
+    - The map might include the hero car as the first entity in entities
+    """
+
+    timestamp: Time = Time()
+    """The timestamp this map was created at.
+
+    Should be the time when this map was initially sent off
+    from the mapping_data_integration node.
+
+    This timestamp is also the "freshest" compared to
+    the timestamps of all entities included in the map
+    """
+    entities: List[Entity] = field(default_factory=list)
+    """The entities this map consists out of
+
+    Note that this list might also include the hero car (as first element of this list)
+    """
+
+    def hero(self) -> Optional[Entity]:
```

Comment:
hero should probably also be a property ??
so user can write map.hero instead of map.hero

just add an @property in line above

Reply:
Would be nice, but cython currently does not properly support the property decorator :-(